frontend/src/queries/Query/Query.tsx

```diff
@@ -71,16 +72,92 @@ export function Query<Q extends Node>(props: QueryProps<Q>): JSX.Element | null
         if (propsQuery !== localQuery) {
             localSetQuery(propsQuery)
         }
-    }, [propsQuery])
+    }, [propsQuery]) // eslint-disable-line react-hooks/exhaustive-deps
 
     const query = readOnly ? propsQuery : localQuery
     const setQuery = propsSetQuery ?? localSetQuery
 
-    const queryContext = props.context || {}
+    const queryContext = useMemo(() => props.context || {}, [props.context])
 
     const uniqueKey =
         props.uniqueKey ?? (props.context?.insightProps && insightVizDataNodeKey(props.context.insightProps))
 
+    const component = useMemo(() => {
```

Comment:
I have the same question here. We're doing more work than the key lookup above, but a few if statements and creating a react element are not expensive computations. Does it prevent some unnecessary re-renders?

Reply:
Done, removed this